Source/ParleyTests/MessagesManagerTests.swift

```diff
@@ -0,0 +1,274 @@
+        XCTAssertEqual(messagesManager.messages[3], secondMessage, "Fourth message should be the second user message.")
+    }
+    
+    func testWelcomeMessage_ShouldBeSetAndAddedToMessages_WhenSettingTheWelcomeMessage() {
+        setWelcomeMessage()
+        messagesManager.loadCachedData()
+        
+        XCTAssertEqual(messagesManager.welcomeMessage, MESSAGE_WELCOME_TEXT)
+        
+        guard let firstMessage = messagesManager.messages.first else {
+            XCTFail("Welcome message should have been added to the array.") ; return
+        }
+        XCTAssertEqual(firstMessage.message, MESSAGE_WELCOME_TEXT, "First message should be the welcome message.")
+        
+        XCTAssertEqual(messagesManager.messages.count, 1, "Welcome message should be the only message in the messages array.")
+    }
+    
+    func testAddUserMessage_ShouldReturnCorrectIndexPath_WhenAddingInfoMessageAndFirstUserMessage() {
+        let firstMessageInsertionIndexPaths = addInfoMessageAndFirstUserMessage()
+        
+        XCTAssertEqual(firstMessageInsertionIndexPaths[0].row, 1, "First index path should correspond to the second item in the table view.")
+        XCTAssertEqual(firstMessageInsertionIndexPaths[0].section, 0, "Section should always to 0.")
+        
+        XCTAssertEqual(firstMessageInsertionIndexPaths[1].row, 2, "Second index path should correspond to the third item in the table view.")
+        XCTAssertEqual(firstMessageInsertionIndexPaths[1].section, 0, "Section should always to 0.")
+        
+        XCTAssertEqual(firstMessageInsertionIndexPaths.count, 2)
+    }
+    
+    func testFirstUserMessage_ShouldAppearAfterWelcomeMessage_WhenAddingUserMessageAfterInfoMessage() {
+        setWelcomeMessage()
+        messagesManager.loadCachedData()
+        
+        let firstMessageText = "Good morning!"
+        let currentDate = Date()
+        let userMessage = createUserMessage(firstMessageText, date: currentDate)
+        
+        _ = messagesManager.add(userMessage)
+        
+        XCTAssertEqual(messagesManager.messages[0].message, MESSAGE_WELCOME_TEXT, "First message should be welcome message.")
+        XCTAssertEqual(messagesManager.messages[1].type, .date, "Second message should be date header.")
+        XCTAssertEqual(messagesManager.messages[1].message, currentDate.asDate(), "Second message should contain the date of the added message.")
+        XCTAssertEqual(messagesManager.messages[2].message, firstMessageText, "Third message should be the added message.")
+        XCTAssertEqual(messagesManager.messages[2].type, .user, "Added message should be from the user.")
+        
+        XCTAssertNil(messagesManager.lastSentMessage, "Message should not be considered as being sent.")
+    }
+    
+    func testTwoMessagesOnDifferentDays_ShouldBeSeparatedByDateIndicators_WhenMessagesAreOnDifferentDays() throws {
```

Comment:
the component name can be just `testTwoMessages` since 'onDifferentDays' is a "when"-reason